### DISCUSSION TITLE
Displaying image thumbnails in dmenu

### DISCUSSION THREAD

**Discussion by kamssiopeia** [davatorium/rofi#2001] [Q&A] [answered]
Hello, I want to create rofi wallpaper picker with image preview. The problem is i can't get rofi to display image preview in `dmenu` like `filebrowser` mode does. Does anyone know how to achieve this?

**Comment by yusufaktepe** [comment]
This is what I'm using with the `fullscreen-preview` theme provided with rofi:
```sh
#!/bin/sh

WallDir=${1:-~/Pictures/Wallpapers}

PREVIEW=true \
rofi -no-config -theme fullscreen-preview.rasi \
	-show filebrowser -filebrowser-command 'setbg' \
	-filebrowser-directory "$WallDir" \
	-filebrowser-sorting-method mtime \
	-selected-row 1 >/dev/null
```

**Reply by DaveDavenport** [reply, marked as answer]
If your rofi is new enough, you can pass filename as icon.  See the rofi-script manpage.


```bash
for a in *.jpg; do echo -en "$a\0icon\x1f$a\n" ; done | rofi -dmenu
```

**Reply by kamssiopeia** [reply]
thank you for answers guys!

edit: I will also post documentation https://github.com/davatorium/rofi/blob/next/doc/rofi-thumbnails.5.markdown
